public/video-ui/src/components/utils/GridImageSelect.js

```diff
@@ -56,56 +56,36 @@ export default class GridEmbedder extends React.Component {
     }
 
     this.closeModal();
-    this.onUpdatePosterImage(data.crop.data);
+    this.onUpdatePosterImage(data.crop.data, data.image);
   };
 
   render() {
-    if (this.props.createMode && this.props.fieldValue) {
-      return (
-        <div className="gridembedder">
-          <div className="gridembedder__button" onClick={this.toggleModal}>
-            <Icon icon="add_to_photos" className="icon__edit" />
-          </div>
-
-          <Modal isOpen={this.state.modalOpen} dismiss={this.closeModal}>
-            <iframe className="gridembedder__modal" src={this.props.gridUrl} />
-          </Modal>
-        </div>
-      );
-    } else if (this.props.createMode) {
-      return (
-        <div className="gridembedder">
-          <div className="gridembedder__noposter" onClick={this.toggleModal}>
-            <div className="gridembedder__noposter-elements">
-              <span>Add Grid Image</span>
-              <div>
-                <Icon icon="add_to_photos" className="icon__edit" />
-              </div>
-            </div>
-          </div>
-
-          <Modal isOpen={this.state.modalOpen} dismiss={this.closeModal}>
-            <iframe className="gridembedder__modal" src={this.props.gridUrl} />
-          </Modal>
-        </div>
-      );
+
+    let gridUrl;
```

Comment:
could this logic be pushed to a function to avoid the `let`?

```js
function getGridUrl() {
  if (this.props.posterImge & this.props.isComposerImage) {
    const imageIdParts = this.props.posterImage.mediaId.split("/");
    const imageGridId = imageIdParts[imageIdParts.length - 1];
    return `${this.props.gridUrl}/images/${imageGridId}/crop?cropType=composer`;
  }
  else {
    return `${this.props.gridUrl}?cropType=video`;
  }
}

const gridUrl = getGridUrl();
```